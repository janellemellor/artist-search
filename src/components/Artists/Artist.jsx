import React from 'react';
import PropTypes from 'prop-types';

const Artist = ({ name }) => (
  <>
    <p>{name}</p>
  </>

);

Artist.propTypes = {
  name: PropTypes.string.isRequired
};

export default Artist;

