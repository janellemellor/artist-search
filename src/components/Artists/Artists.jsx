// import React from 'react';
import Artist from '../Artists/Artist.jsx';
import withList from '../withLists.js';

const Artists = withList(Artist, 'artistResults');

export default Artists;
