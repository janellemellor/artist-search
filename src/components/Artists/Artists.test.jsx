import React from 'react';
import { shallow } from 'enzyme';
import Artists from '../Artists/Artists.jsx';

describe('Artists', () => {
  it('matches the Artists snapshot', () => {
    const artistResults = [
      {
        id: 'a3cb23fc-acd3-4ce0-8f36-1e5aa6a18432',
        name: 'U2'
      },
      {
        id: 'b613c677-42e5-47e8-abcd-820c48f17447',
        name: 'U2 Zoo'
      }
    ];

    const wrapper = shallow(<Artists artistResults={artistResults} />);
    expect(wrapper).toMatchSnapshot();
  });
});

