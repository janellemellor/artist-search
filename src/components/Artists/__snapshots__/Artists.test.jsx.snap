// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists matches the Artists snapshot 1`] = `
<ul>
  <li
    key="a3cb23fc-acd3-4ce0-8f36-1e5aa6a18432"
  >
    <Link
      to="/a3cb23fc-acd3-4ce0-8f36-1e5aa6a18432"
    >
      <Artist
        id="a3cb23fc-acd3-4ce0-8f36-1e5aa6a18432"
        name="U2"
      />
    </Link>
  </li>
  <li
    key="b613c677-42e5-47e8-abcd-820c48f17447"
  >
    <Link
      to="/b613c677-42e5-47e8-abcd-820c48f17447"
    >
      <Artist
        id="b613c677-42e5-47e8-abcd-820c48f17447"
        name="U2 Zoo"
      />
    </Link>
  </li>
</ul>
`;
